src/pages/ShoppingList.js

```diff
@@ -84,26 +101,50 @@ const ShoppingList = ({ token }) => {
             getShoppingList();
         });
     };
+
+    const listFilterChange = e => {
+        let normalizeString = e.target.value
+            .toLowerCase()
+            .replace(/[.,\/#!$%\^&\*;:{}=\-_`~()]/g, '')
+            .trim()
+            .replace(/\s{2,}/g, ' ');
+        setFilterString(normalizeString);
+    };
```

Comment:
We've used this normalizing pattern a few times now in the app. I wonder if we could extract the logic out into a utility function to be re-used throughout. Just something to consider, since it's proven useful.

Reply:
We were having a conversation about the exact same thing earlier this week! I'll make a note to bring it up during the meeting.